src/codegen/compile_classical.ml

```diff
@@ -8309,7 +8324,13 @@ module MakeSerialization (Strm : Stream) = struct
       ReadBuf.set_size get_ref_buf (get_refs_size ^^ compile_mul_const Heap.word_size) ^^
 
       (* Go! *)
-      Bool.lit extended ^^ get_data_buf ^^ get_typtbl_ptr ^^ get_typtbl_size_ptr ^^ get_maintyps_ptr ^^
+      let tydesc, _, _ = type_desc env ts in
+      let tydesc_len = Int32.of_int (String.length tydesc) in
+      let tydesc_tolerance =
+        compile_unboxed_const tydesc_len ^^
```

Comment:
Is it possible for this overflow and then too small a limit is applied?
Assuming 10MB max payload  (for intercanister messages) then we have 10M bytes 
and  Uint32.max is 4,294,967,295, so that would require a very large bias or coefficient > 42?
But check my arithmetic... 

Maybe not worth worrying about.

Reply:
My arithmetic sucks and you are quite right